staging/src/k8s.io/client-go/tools/record/events_cache.go

```diff
@@ -134,8 +131,6 @@ func (f *EventSourceObjectSpamFilter) Filter(event *v1.Event) bool {
 	eventKey := f.spamKeyFunc(event)
 
 	// do we have a record of similar events in our cache?
-	f.Lock()
-	defer f.Unlock()
```

Comment:
this is guarding more than just the cache get/add ... also the rate limiter assignment to the record (which is not thread-safe, and ensuring we don't get concurrent accept calls to the record's rate limiter

Reply:
the golang rate limiter is thread safe, that is wrapped by us